packages/battery/test/battery_test.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:async/async.dart';
+import 'package:flutter/services.dart';
+import 'package:test/test.dart';
+import 'package:battery/battery.dart';
+import 'package:mockito/mockito.dart';
+
+void main() {
+  MockMethodChannel methodChannel;
+  MockEventChannel eventChannel;
+  Battery battery;
+
+  setUp(() {
+    methodChannel = new MockMethodChannel();
+    eventChannel = new MockEventChannel();
+    battery = new Battery.private(methodChannel, eventChannel);
```

Comment:
[nit] The mockery shouldn't be necessary. In particular, there should be no need to expose channel use from your production code.

Channels are completely stateless objects and don't keep references to their method call handlers. This means you can test every aspect of the communication behavior of `Battery` just by reusing the channel name in another channel object, created by your test. Then call `batteryTestChannel.setMockMethodCallHandler` in place of recording mock expectations to test proper calling and result handling for outgoing method calls. Or use `BinaryMessages.handlePlatformMessage` for simulating incoming events on the event channel.